### PR TITLE
Reject range "don't care" match fields in Stratum-bfrt

### PR DESCRIPTION
**Pull request by pudelkoM**
According to the [P4Runtime spec](https://opennetworking.org/wp-content/uploads/2020/10/P4Runtime-Specification-120-wd.html#sec-match-format) "don't care" range match field must be omitted from a request. This change enables a check to ensure that a clients does not send those.

### CODE REVIEW

File: stratum/hal/lib/barefoot/bfrt_table_manager.cc [hunk 1]
```diff
@@ -176,7 +176,9 @@ ::util::Status BfrtTableManager::BuildTableKey(
                                 ::p4::config::v1::MatchField::EXACT)
               << "Found match field of type EXACT does not fit match field "
               << expected_match_field.ShortDebugString() << ".";
-          CHECK_RETURN_IF_FALSE(!IsDontCareMatch(mk.exact()));
```
Comment (pierventre) [inline stratum/hal/lib/barefoot/bfrt_table_manager.cc:179]: I believe there is no "don't care" for exact match

**Review by pierventre — APPROVED** [review]
the implementation for exact match returns always false but it is quite misleading. Up to you.

File: stratum/hal/lib/barefoot/bfrt_table_manager.cc (hunk 1 above)
Comment (pudelkoM) [reply stratum/hal/lib/barefoot/bfrt_table_manager.cc:179]: correct, that function will always return false.

Comment (pierventre) [reply stratum/hal/lib/barefoot/bfrt_table_manager.cc:179]: maybe add a comment :)

**Pull request merged**